internal/cmdrender/executor.go

```diff
@@ -40,22 +41,35 @@ import (
 
 var errAllowedExecNotSpecified error = fmt.Errorf("must run with `--allow-exec` option to allow running function binaries")
 
-// Executor hydrates a given pkg.
-type Executor struct {
-	PkgPath         string
-	ResultsDirPath  string
-	Output          io.Writer
+// Renderer hydrates a given pkg by running the functions in the input pipeline
+type Renderer struct {
+	// PkgPath is the absolute path to the root package
+	PkgPath string
+
+	// ResultsDirPath is absolute path to the directory to write results
+	ResultsDirPath string
+
+	// Output is the writer to which the output resources are written
+	Output io.Writer
+
+	// ImagePullPolicy pull image before running the container.
+	// It must be one of fnruntime.AlwaysPull, fnruntime.IfNotPresentPull, fnruntime.NeverPull
 	ImagePullPolicy fnruntime.ImagePullPolicy
-	AllowExec       bool
+
+	// AllowExec allow binary executable to be run during pipeline execution
+	AllowExec bool
+
+	// FileSystem is the input filesystem to operate on
+	FileSystem filesys.FileSystem
 }
 
 // Execute runs a pipeline.
-func (e *Executor) Execute(ctx context.Context) error {
+func (e *Renderer) Execute(ctx context.Context) error {
```

Comment:
nit: we should rename `Execute` to `Render` and this will effectively the method that satisfy the Renderer interface in `pkg/fn`

Reply:
Done. I moved it to util/render to be consistent with other apis.